pkg/certificates/config_test.go

```diff
@@ -0,0 +1,433 @@
+			expectError: false,
+		},
+		{
+			name: "missing cluster name",
+			config: &RenewalConfig{
+				ControlPlane: NodeConfig{
+					Nodes:   []string{"192.168.1.10"},
+					OS:      "ubuntu",
+					SSHKey:  "/tmp/test-key",
+					SSHUser: "ec2-user",
+				},
+			},
+			expectError: true,
+			errorMsg:    "cluster name is required",
+		},
+		{
+			name: "missing control plane nodes",
+			config: &RenewalConfig{
+				ClusterName: "test-cluster",
+				ControlPlane: NodeConfig{
+					OS:      "ubuntu",
+					SSHKey:  "/tmp/test-key",
+					SSHUser: "ec2-user",
+				},
+			},
+			expectError: true,
+			errorMsg:    "at least one node is required",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			runValidateConfigTest(t, tt)
+		})
+	}
+}
+
+// TestSSHKeyNotFound tests the validateNodeConfig function with a non-existent SSH key file.
+func TestSSHKeyNotFound(t *testing.T) {
+	config := NodeConfig{
+		Nodes:   []string{"192.168.1.10"},
+		OS:      "ubuntu",
+		SSHKey:  "/tmp/non-existent-key",
+		SSHUser: "ec2-user",
+	}
+
+	err := validateNodeConfig(&config)
+	if err == nil {
+		t.Error("expected error for non-existent SSH key file but got none")
+	}
+	if !strings.Contains(err.Error(), "retrieving SSH key file information") {
+		t.Errorf("expected error message to contain 'retrieving SSH key file information', got %q", err.Error())
+	}
+}
+
+// parseConfigTestCase defines a test case for ParseConfig.
+type parseConfigTestCase struct {
+	name        string
+	configYaml  string
+	expectError bool
+	errorMsg    string
+}
+
+// createTempConfigFile creates a temporary config file with the given YAML content.
+func createTempConfigFile(t *testing.T, content string) string {
+	tmpfile, err := os.CreateTemp("", "config-*.yaml")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if _, err := tmpfile.Write([]byte(content)); err != nil {
+		t.Fatal(err)
+	}
+	if err := tmpfile.Close(); err != nil {
+		t.Fatal(err)
+	}
+
+	return tmpfile.Name()
+}
+
+// runParseConfigTest runs a single ParseConfig test case.
+func runParseConfigTest(t *testing.T, tt parseConfigTestCase) {
+	// Create temporary config file
+	configFile := createTempConfigFile(t, tt.configYaml)
+	defer os.Remove(configFile)
+
+	// Create temporary SSH key file
+	keyFile := "/tmp/test-key"
+	cleanup := setupSSHKeyForTest(t, keyFile)
+	defer cleanup()
+
+	// Test config parsing
+	_, err := ParseConfig(configFile)
+	if tt.expectError && err == nil {
+		t.Error("expected error but got none")
+	}
+	if !tt.expectError && err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	if tt.expectError && err != nil && tt.errorMsg != "" && !contains(err.Error(), tt.errorMsg) {
+		t.Errorf("expected error message to contain %q, got %q", tt.errorMsg, err.Error())
+	}
+}
+
+// getParseConfigTestCases returns test cases for ParseConfig.
+func getParseConfigTestCases() []parseConfigTestCase {
+	return []parseConfigTestCase{
+		{
+			name: "valid config with both etcd and control plane",
```

Comment:
These seem redundant test cases, since we are verifying all these test cases under TestValidateConfig, can we only have which are needed to verify https://github.com/aws/eks-anywhere/blob/main/pkg/certificates/config.go#L26?